compiler/rustc_mir/src/transform/check_unsafety.rs

```diff
@@ -132,6 +132,24 @@ impl<'a, 'tcx> Visitor<'tcx> for UnsafetyChecker<'a, 'tcx> {
                     self.register_violations(&violations, &unsafe_blocks);
                 }
             },
+            Rvalue::Cast(kind, source, mir_cast_ty) => {
+                if let CastKind::Pointer(ty::adjustment::PointerCast::Unsize) = kind {
```

Comment:
So `&dyn Foo` -> `&dyn Bar` casts are also "unsize" casts (even though they do not literally go from `Sized` to `!Sized`)?

Reply:
I think so, but I was just following the existing infrastructure within the compiler that wasn't in use before my previous PRs. I'll refactor if it's decided that this be called something else.

Reply:
Okay, makes sense.